cargo-dist/templates/installer/homebrew.rb.j2

```diff
@@ -68,15 +68,29 @@ class {{ formula_class }} < Formula
 
   def install
     {#- Like the URL case above, write out a single install line in the case that the binary artifacts are the same across architectures #}
-    {%- if arm64_macos.binaries == x86_64_macos.binaries %}
-    bin.install {% for binary in arm64_macos.binaries %}"{{ binary }}"{{ ", " if not loop.last else "" }}{% endfor %}
-    {%- else %}
-    if Hardware::CPU.type == :arm
+    on_macos do
```

Comment:
If we're going to introduce per-platform blocks, I think we need to guard against adding the block at all if the platform is missing. Eg if there are only Linux binaries, we shouldn't have an `on_macos` block, and vice versa.

Reply:
It's not actually about different names; rather, it's possible for a project with *multiple* binaries to only build some of them on certain platforms. It's a niche usecase, however.

Reply:
Thanks. It certainly makes this much harder. Do you have a preferred way to approach fixing this?

In my testing, having an empty stanza (say for ARM64 Linux when on x86_64 Linux) was not an issue.